library/universes.mli

```diff
@@ -243,28 +243,31 @@ val choose_canonical : universe_set -> (Level.t -> bool) (* flexibles *) -> univ
     
 val compute_lbound : (constraint_type * Univ.universe) list -> universe option
 
+type lowermap = constraint_type Univ.LMap.t
+
 val instantiate_with_lbound : 
   Univ.LMap.key ->
   Univ.universe ->
+  lowermap ->
   bool ->
   bool ->
   Univ.LSet.t * Univ.universe option Univ.LMap.t *
     Univ.LSet.t *
-    (bool * bool * Univ.universe) Univ.LMap.t * Univ.constraints ->
+    (bool * bool * Univ.universe * lowermap) Univ.LMap.t * Univ.constraints ->
   (Univ.LSet.t * Univ.universe option Univ.LMap.t *
     Univ.LSet.t *
-     (bool * bool * Univ.universe) Univ.LMap.t * Univ.constraints) *
-    (bool * bool * Univ.universe)
+     (bool * bool * Univ.universe * lowermap) Univ.LMap.t * Univ.constraints) *
+    (bool * bool * Univ.universe * lowermap)
```

Comment:
Please give a type name to these quadruplets. It would be even better if you could turn them into records.